bugtool/cmd/root.go

```diff
@@ -0,0 +1,297 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/daemon/defaults"
+
+	"github.com/pierrre/archivefile/zip"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// TODO: do we need filtering options?
+
+var RootCmd = &cobra.Command{
+	Use:   "bugtool",
+	Short: "retrieve information and files for debugging single Cilium node",
+	Long:  "TODO",
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("bugtool")
+		runTool()
+	},
+}
+
+var (
+	serve    bool
+	port     int
+	dumpPath string
+)
+
+func init() {
+	RootCmd.Flags().BoolVar(&serve, "serve", false, "Serve captured files over HTTP")
+	RootCmd.Flags().IntVarP(&port, "port", "p",
+		4444, "Port to use when serving files, default is 4444")
+	RootCmd.Flags().StringVarP(&dumpPath, "tmp", "t", "/tmp/", "Path to store extraced files")
+}
+
+func runTool() {
+	var dbgDir string
+	// Prevent collision with other directories
+	d, err := ioutil.TempDir(dumpPath, "bugtool-")
+	if err != nil {
+		log.WithError(err).Warn("failed to create debug directory")
+	}
+	dbgDir = d
+
+	// TODO: review all error messages.
+	copyStateDir(dbgDir)
+	copySystemInfo(dbgDir)
+	// TODO: Copy relevent log files
+	// TODO: retrieve k8s log files
+
+	/*
+		Docker logs
+		access logs
+	*/
+
+	copyResolveConfig(dbgDir)
+	copyKernelConfig(dbgDir)
+	createArchive(dbgDir, dumpPath)
+
+	if serve {
+		serveStaticFiles(dbgDir)
+		return
```

Comment:
Is it intentional to keep the tmp directory around after serving?

Reply:
Yes was mostly to check files we in there, but removed.